Feature/kas 3460 improve agenda search ux

sergiofenoll opened the pull request:
Adds inline loading, so that the search input does not get re-rendered while searching (and the user does not lose focus).
Adds buttons to explicitly start the search.
Keeps support for debounced search-as-you-type (wasn't sure to me if that was expected, but it seems useful to me), but typing Enter or clicking the search button performs the search immediately.
Uses `AuButton` for the search buttons instead of `Auk::Button`

Note: the icon size in `AuButton` looks somewhat small. Perhaps we want to make it larger with CSS overrides. Or maybe we could add an `@iconSize` argument to the `AuButton` component in https://github.com/appuniversum/ember-appuniversum (an `AuIcon` with `@size="large"` looks okay in our buttons).
![image](https://user-images.githubusercontent.com/22411874/175921001-093206b8-21d0-4ff8-af1c-c00c193960f5.png)

sergiofenoll committed:
feat: add inline loading state for agendaitem search

sergiofenoll committed:
feat: add buttons to agendaitem search

The debounced search that was part of this component is still available,
so we will search as a user types, but if a user either presses enter or
clicks on the search button, we will execute the search immediately

sergiofenoll committed:
feat: use AuButton instead of Auk::Button

erikap commented on app/routes/agenda/agendaitems.js:
```diff
@@ -117,6 +117,24 @@ export default class AgendaAgendaitemsRoute extends Route {
+      controller.isLoadingModel = false;
+    });
+
+   // only bubble loading event when transitioning between tabs
+   // to enable loading template to be shown
```
Is this comment up-to-date?  'tabs' are not applicable here? 
I assume we want to say "don't bubble loading event if only query params change"?

sergiofenoll replied:
I just copy-pasted this snippet to be honest, but I think it still applies? We still want to show the template-level loading state when switching between the Overzicht, Detail, Vergelijken, and Documenten tab, right?

But the comment is somewhat confusing and really just an artifact of being copied around, I will try to make it a bit clearer

erikap commented on app/templates/agenda/agendaitems.hbs:
```diff
@@ -33,6 +37,7 @@
       <Agenda::AgendaitemSearch
         @searchText={{readonly this.filter}}
         @onSearch={{this.searchAgendaitems}}
+        @fullwidth={{true}}
```
`fullwidth` doesn't clearly reflect the intention of the argument for me. Maybe name it in line with the argument of `AuButton`?  `hideButtonText` or something like that?

sergiofenoll committed:
docs: improve comment about loading event bubbling in agendaitems route

sergiofenoll committed:
style: rename fullwidth param to hideButtonText in Agenda::AgendaitemSearch

erikap commented:
> Note: the icon size in `AuButton` looks somewhat small. Perhaps we want to make it larger with CSS overrides. Or maybe we could add an `@iconSize` argument to the `AuButton` component in https://github.com/appuniversum/ember-appuniversum (an `AuIcon` with `@size="large"` looks okay in our buttons). ![image](https://user-images.githubusercontent.com/22411874/175921001-093206b8-21d0-4ff8-af1c-c00c193960f5.png)

@brenner-company What's your opinion about the icon-size? What's the best fix? Do we create a follow-up story for this or block this story?

sergiofenoll committed:
fix: stop loading event bubbling only when filter changed

brenner-company commented:
@erikap @sergiofenoll I'm inclined to switching the button back to the auk-version for the time being because:
- The component is, as you are stating, missing some styling overrides like icon size (I wouldn't go for 'large' size assuming it should be correctly sized by default), focus color, button height, etc.
- The component is not 100% complete for use within Kaleidos regarding available skins (missing 'success', etc.).

Normally I would pick this up along my other work but I'm on my last day before a rather long vacation 🏖. So a quick follow-up is not possible from my side I'm afraid.

brenner-company commented:
@sergiofenoll Thank you for you initiative by the way in using the components from `ember-appuniversum` 😊

sergiofenoll committed:
fix: use Auk::Button instead of AuButton for agendaitem search

erikap committed:
Merge branch 'development' into feature/KAS-3460-improve-agenda-search-ux

sergiofenoll committed:
fix: replace misplaced double quote

sergiofenoll committed:
test: wait after searching agendaitems before looking for results

Searching for agendaitems no longer forces a page refresh. This means
that in the tests the expectation that the results will be rendered as
soon as the network call has happened no longer holds.
